### PR TITLE
Api errors

### LINKED ISSUE #116: Return JSON Content-Type for error messages
To be able to easily parse error messages it might be a good idea to return errors also as JSON Content-Type.
Currently you only get `text/html` as response Content-Type (at least for 404 status code). I would expect when I set `Accept: application/json` to also get a json error message.

### LINKED ISSUE #134: Return meaningful error messages
## Problem
Viele Menschen beschweren sich (und viele tun brechen frustriert ab ohne Rückmeldung zu geben) etwa wie folgt: ich habe einen Termin versucht einzugeben und es ging nicht, es kam die Fehlermeldung "bad request", das sagt mir nicht, was ich da falsch gemacht habe und ändern muß. Sowas darf nicht passieren, das schreckt die Leute total ab. Ich hab es dann aufgegeben.

## Solution
Error-Messanges is needed for
- Warning that a field is used wrong (which field contains an error)
- If a protected tag is used (only for events)

### PR DESCRIPTION
**Pull request by magdaddy**

### CODE REVIEW

**Comment by uklotzde** [comment]
The CI build builds each sub-crate in isolation. This is different from building the whole workspace and might detect errors in the sub-crates.

Please fix and then force push.

File: src/ports/web/api/mod.rs [hunk 1]
```diff
@@ -443,26 +450,28 @@ fn entries_csv_export(
...
-                            Status::Unauthorized
+                            json_error_response(req, err, Status::Unauthorized)
                         }
-                        ParameterError::UserExists => <Status>::new(400, "UserExists"),
-                        ParameterError::EmailNotConfirmed => {
-                            <Status>::new(403, "EmailNotConfirmed")
+                        // ParameterError::UserExists => Err(<Status>::new(400, "UserExists")),
+                        // ParameterError::EmailNotConfirmed => {
+                        //     Err(<Status>::new(403, "EmailNotConfirmed"))
+                        // }
```
Comment (flosse) [inline src/ports/web/api/mod.rs]: Please either delete this or add a TODO comment with some explanations.

**Review by flosse — CHANGES_REQUESTED** [review]

**Comment by magdaddy** [comment]
After implementing `From<Url> for String` and fixing countless clippy warnings (all of which, by the way, have nothing to do with my PR whatsoever) now clippy gives this:

```
error: couldn't read src/ports/web/frontend/../../../../ofdb-app-clearance/pkg/clearance.js: No such file or directory (os error 2)
  --> src/ports/web/frontend/mod.rs:33:28
   |
33 | const CLEARANCE_JS: &str = include_str!("../../../../ofdb-app-clearance/pkg/clearance.js");
   |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `include_str` (in Nightly builds, run with -Z macro-backtrace for more info)

error: couldn't read src/ports/web/frontend/../../../../ofdb-app-clearance/pkg/clearance_bg.wasm: No such file or directory (os error 2)
  --> src/ports/web/frontend/mod.rs:35:5
   |
35 |     include_bytes!("../../../../ofdb-app-clearance/pkg/clearance_bg.wasm");
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `include_bytes` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 2 previous errors
```

I have no idea how to fix this, but I get a faint memory that you once had something like this @flosse ?

**Comment by flosse** [comment]
That's wired I can't reproduce this locally...

**Review by flosse — APPROVED** [review]

**Comment by flosse** [comment]
... I think we can ignore the clippy warnings for now.

**Pull request merged**